docs/plugin-protocol/tfplugin5.4.proto

```diff
@@ -128,8 +128,34 @@ message Schema {
     Block block = 2;
 }
 
+// ServerCapabilities allows providers to communicate extra information
+// regarding supported protocol features. This is used to indicate
+// availability of certain forward-compatible changes which may be optional
+// in a major protocol version, but cannot be tested for directly.
+message ServerCapabilities {
+    // The plan_destroy capability signals that a provider expects a call
+    // to PlanResourceChange when a resource is going to be destroyed.
+    bool plan_destroy = 1;
+
+    // The get_provider_schema_optional capability indicates that this
+    // provider does not require calling GetProviderSchema to operate
+    // normally, and the caller can used a cached copy of the provider's
+    // schema.
+    bool get_provider_schema_optional = 2;
+}
+
 service Provider {
     //////// Information about what a provider supports/expects
+
+    // GetMetadata returns upfront information about server capabilities and
+    // supported resource types without requiring the server to instantiate all
+    // schema information, which may be memory intensive. This RPC is a later
```

Comment:
Rather than "a later addition" which doesn't explain "later than what?", maybe describe it as an "optional" method, and clients must be able to handle an unimplemented error?

Reply:
Sounds good, will update 👍